src/commands/up.rs

```diff
@@ -148,32 +148,21 @@ impl UpCommand {
             ApplicationTrigger::Redis(_) => "redis",
         };
 
-        // Build and write app lock file
-        let locked_app = spin_trigger::locked::build_locked_app(app, &working_dir)?;
-        let locked_path = working_dir.join("spin.lock");
-        let locked_app_contents =
-            serde_json::to_vec_pretty(&locked_app).context("failed to serialize locked app")?;
-        std::fs::write(&locked_path, locked_app_contents)
-            .with_context(|| format!("failed to write {:?}", locked_path))?;
-        let locked_url = Url::from_file_path(&locked_path)
-            .map_err(|_| anyhow!("cannot convert to file URL: {locked_path:?}"))?
-            .to_string();
-
-        // For `spin up --help`, we just want the executor to dump its own argument usage info
-        let trigger_args = if self.help {
-            vec![OsString::from("--help-args-only")]
-        } else {
-            self.trigger_args
-        };
-
         // The docs for `current_exe` warn that this may be insecure because it could be executed
         // via hard-link. I think it should be fine as long as we aren't `setuid`ing this binary.
         let mut cmd = std::process::Command::new(std::env::current_exe().unwrap());
         cmd.arg("trigger")
-            .env(SPIN_WORKING_DIR, working_dir)
-            .env(SPIN_LOCKED_URL, locked_url)
             .arg(trigger_type)
-            .args(trigger_args);
+            .env(SPIN_WORKING_DIR, &working_dir);
+
+        if self.help {
+            let help_args = vec![OsString::from("--help-args-only")];
+            cmd.args(help_args);
```

Comment:
Can simplify since it doesn't have to match the type of `trigger_args`:

```suggestion
            cmd.arg("--help-args-only");
```